liberty-ls/src/main/java/io/openliberty/tools/langserver/utils/Messages.java

```diff
@@ -80,10 +82,22 @@ public static List<String> getMatchingKeys(String query, LibertyTextDocument tex
         }
         
         String filename = textDocument.getUri();
+        // remove completion results that don't contain the query string (case-insensitive search)
```

Comment:
I don't really follow what this is for. I read the comment, but that is not telling me what the query string is for (example of something we are removing). What is the use case?

Reply:
Some background... the initial implementation never filtered any results for completion, just returned the entire catalog of keys, and let the client-side IDE handle the filtering based on what was typed. 
When it came to running tests, each completion request would return that full list, making the tests pretty useless other than checking that: yes, we return all the possible keys for this file.

Reply:
Functionally when running the IDE extension with the language server, it doesn't seem to make much of a difference. But adding this logic to filter the results on the server-side made it a lot easier to test reasonable functionality for the completion feature.

Reply:
So it filters out based on a key? Or it is filtering out bootstrap vs server.env? Still not clear.

Reply:
Yes, it filters the available keys based on the entered text (query).